### PR TITLE
Deeplink refactor

### PR DESCRIPTION
**Pull request by juanlaube**
## What's this?
This decouples the deeplink parsing from actually handling each url.

### Relationships

Refs #5180, doesn't close because needs some 🤖 bits.

## Why do we want this?
Deeplinking is completely different on iOS and Android, this moves the handling to the platform code and we have extension methods for everything we can share.

## :squid: Permissions
Only mvexit bois.

### CODE REVIEW

File: Toggl.iOS/Startup/AppDelegate.Deeplink.cs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        private async Task showNewTimeEntry(DeeplinkParameters.TimeEntryParameters timeEntry, ITimeService timeService, INavigationService navigationService)
+        {
+            var workspaceId = timeEntry.WorkspaceId;
+            var startTime = timeEntry.StartTime ?? timeService.CurrentDateTime;
+            var duration = timeEntry.Duration;
+            var description = timeEntry.Description;
+            var projectId = timeEntry.ProjectId;
+            var tags = timeEntry.TagIds;
+
+            var startTimeEntryParameters = new StartTimeEntryParameters(
```
Comment (daividssilverio) [inline Toggl.iOS/Startup/AppDelegate.Deeplink.cs]: Can we add another extension method to convert the
`DeeplinkParameters.TimeEntryParameters` into `StartTimeEntryParameters`?

Comment (juanlaube) [reply Toggl.iOS/Startup/AppDelegate.Deeplink.cs]: Done!

File: Toggl.Core.UI/Parameters/DeeplinkParameters.cs [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+                    matchEditTimeEntry(editTimeEntryParameters);
+                    return;
+                case DeeplinkShowReportsParameters showReportsParameters:
+                    matchShowReports(showReportsParameters);
+                    return;
+                case DeeplinkShowCalendarParameters showCalendarParameters:
+                    matchShowCalendar(showCalendarParameters);
+                    return;
+                default:
+                    matchOther(this);
```
Comment (daividssilverio) [inline Toggl.Core.UI/Parameters/DeeplinkParameters.cs]: Won't this crash by default? The default is null. The iOS implementation isn't passing any value for this.

File: Toggl.Core.UI/Parameters/DeeplinkParameters.cs [hunk 3]
```diff
@@ -120,9 +120,11 @@ public static DeeplinkParameters WithCalendar(string eventId)
...
                     matchShowCalendar(showCalendarParameters);
                     return;
-                default:
-                    matchOther(this);
-                    return;
+            }
+
+            if (matchOther != null)
+            {
+                matchOther(this);
```
Comment (heytherewill) [inline Toggl.Core.UI/Parameters/DeeplinkParameters.cs]: `matchOther?.Invoke(this)`

**Pull request merged**